### PR TITLE
docs: add TPU/UDP protocol spec

### PR DESCRIPTION
**Pull request by riptl**

### CODE REVIEW

File: protocol/p2p/tpu.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+Nodes expose multiple TPU endpoints to separate classes of transactions into distinct network flows.
+
+- **TPU** is the default endpoint permitting unauthenticated clients
+- **TPUvote** is reserved for Tower BFT vote transactions from staked validators
+- **TPUfwd** ingests unprocessed transactions from the previous slot leader and inhibits further forwarding
+
+TPU/UDP protocol
+----------------
+
+TPU/UDP is a simple datagram-oriented network protocol.
```
Comment (riptl) [inline protocol/p2p/tpu.md:64]: should mention MTU

File: protocol/p2p/tpu.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+Signers can either submit transactions directly to TPU nodes (requires UDP connectivity) or submit them through RPC servers.
+
+In a Solana cluster, full nodes usually participate as TPU relayers, although not required.
+All validators included in the leader schedule should operate an open TPU service.
+
+Nodes attempt to forward transactions to the current leader's TPU endpoints.
+The node identities of the leader schedule are well known (runtime entrinsic),
+and their TPU endpoints are part of CRDS ContactInfo.
+
+The network leader ingests TPU traffic into its transaction processing unit.
```
Comment (leoluk) [inline protocol/p2p/tpu.md]: (while it is leader, acts as relayer otherwise?)

File: protocol/p2p/tpu.md (hunk 1 above)
Comment (leoluk) [reply protocol/p2p/tpu.md:64]: No longer matters with QUIC though - we can just drop the whole section?

File: protocol/p2p/tpu.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+-----------------
+
+:::todo
+document TLS preferences, streames, frames
+:::
+
+Security
+--------
+
+TPU traffic should be minimally exposed.
```
Comment (leoluk) [inline protocol/p2p/tpu.md]: Maybe clarify this as "TPU traffic should be treated as confidential data"

**Review by leoluk — APPROVED** [review]

File: protocol/p2p/tpu.md (hunk 1 above)
Comment (riptl) [reply protocol/p2p/tpu.md:64]: I'd prefer to leave it there for completeness, and at least until no more nodes (on devnet, testnet, mainnet, pythnet) are using it. Let's put TPU/QUIC before TPU/UDP though

**Pull request merged**